src/Xenial.Tasty.Cli/Program.cs

```diff
@@ -0,0 +1,236 @@
+            => Promise((resolve) =>
+            {
+                Func<Task> cancelKeyPress = () => Promise((resolve, reject) =>
+               {
+                   Console.CancelKeyPress += (_, e) =>
+                   {
+                       if (e.Cancel)
+                       {
+                           Console.WriteLine("Cancelling execution...");
+                           reject();
+                       }
+                   };
+               });
+
+                Func<Task> endTestPipelineSignaled = () => Promise((resolve, reject) =>
+                {
+                    tastyServer.EndTestPipelineSignaled = () =>
+                    {
+                        Console.WriteLine("Cancelling execution...");
+                        reject();
+                    };
+                });
+
+                Func<Task<ConsoleKeyInfo>> readConsoleKey = () => Promise<ConsoleKeyInfo>(resolve =>
+                {
+                    _ = Task.Run(() =>
+                    {
+                        var info = Console.ReadKey(true);
+                        resolve(info);
+                    });
+                });
+
+                Action writeCommands = () =>
+                {
+                    Console.WriteLine("Interactive Mode");
+                    Console.WriteLine("================");
+
+                    foreach (var c in commands)
+                    {
+                        Console.WriteLine($"{c.Name} - {c.Description}" + (c.IsDefault ? " (default)" : string.Empty));
+                    }
+
+                    Console.WriteLine("c - Cancel");
+                    Console.WriteLine("================");
+                };
+
+                Func<ConsoleKeyInfo, SerializableTastyCommand?> findCommand = (info) =>
+                {
+                    var key = info.Key.ToString().ToLower();
+
+                    var command = info.Key == ConsoleKey.Enter
+                        ? commands.FirstOrDefault(c => c.IsDefault)
+                        : commands.FirstOrDefault(c => c.Name.ToLowerInvariant() == key);
+
+                    return command;
+                };
+
+                Func<Task<Func<Task>>> readInput = async () =>
+                {
+                    var info = await readConsoleKey();
+                    var command = findCommand(info);
+                    if (command != null)
+                    {
+                        return async () =>
+                        {
+                            Console.WriteLine($"Executing {command.Name} - {command.Description}");
+
+                            await tastyServer.DoExecuteCommand(new ExecuteCommandEventArgs
+                            {
+                                CommandName = command.Name
+                            });
+                        };
+                    }
+                    if (info.Key == ConsoleKey.C)
```

Comment:
Only "C" not Ctrl+C?

Reply:
Ctrl+C is covered with `Console.CancelKeyPress` and should be a kill switch. `C` should be able to kill the testrun at any moment without killing CLI it self. (At least in theory)